### PR TITLE
ClientForwardController Change Regex Mapping

### LINKED ISSUE #18039: ClientForwardController intercepts request but he should not
Hi,

I think, I found a minor issue in ClientForwardController. By the way JHipster rocks !

##### **Overview of the issue**

    @GetMapping(value = "/**/{path:[^\\.]*}")
With that configuration of ClientForwardController,

And with a controller with this configuration

@RequestMapping("/api")

...

`@GetMapping("/{one}/{two}/{three}/{four}/{five}")
    public ResponseEntity<String> getValue(
        @PathVariable(name = "one") int one,
        @PathVariable(name = "two") int two,
        @PathVariable(name = "three") int three,
        @PathVariable(name = "four") int four,
        @PathVariable(name = "five") int five
    )  {
        return ResponseEntity.ok("hello world");
    }`

With this request : http://localhost:8080/api/1/2/3/4/5

I don't enter into my function getValue, it's ClientForwardController which intercepts the request.

If i remove "five" pathVariable, I enter into the function getValue. a bit strange !

##### **Suggest a Fix**

    @GetMapping(value = "/{path:[^\\.]*}")

Remove "/**" at start like the case communicationSpringWebsocket in ClientForwardController.java.ejs and getValue works with 5 pathVariable

### PR DESCRIPTION
**Pull request by maiido**
closes #18039

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/web/rest/ClientForwardController.java.ejs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public class ClientForwardController {
 <%_ if (communicationSpringWebsocket) { _%>
     @GetMapping(value = {"/{path:[^\\.]*}", "/{path:^(?!websocket).*}/**/{path:[^\\.]*}"})
 <%_ } else { _%>
-    @GetMapping(value = "/**/{path:[^\\.]*}")
+    @GetMapping(value = "/{path:[^\\.]*}")
```
Comment (vishal423) [inline generators/server/templates/src/main/java/package/web/rest/ClientForwardController.java.ejs:34]: have you tested with base path?

Comment (maiido) [reply generators/server/templates/src/main/java/package/web/rest/ClientForwardController.java.ejs:34]: sorry, not sure to understand the question

Comment (pascalgrimaud) [reply generators/server/templates/src/main/java/package/web/rest/ClientForwardController.java.ejs:34]: I think @vishal423 want you to test with base path, it means: localhost:8080/xxxx/

But honestly, it's just a pain to configure all the project, to make it work.

So for me, it's the correct fix and people who uses base path can test and propose an improvement here

**Comment by pascalgrimaud** [comment]
@maiido : can you update the test plz? Otherwise the CI will fail

**Comment by OmarHawk** [comment]
> @maiido : can you update the test plz? Otherwise the CI will fail

I have tested this change in a built application and this makes the admin interface unavailable - if we push f5 or use a direct link... so please also make sure that stuff like that will continue to work. ;-)

**Comment by maiido** [comment]
> > @maiido : can you update the test plz? Otherwise the CI will fail
> 
> I have tested this change in a built application and this makes the admin interface unavailable - if we push f5 or use a direct link... so please also make sure that stuff like that will continue to work. ;-)

Hi, it's strange, I don't have this issue. My test : 
Compile with maven (profiles dev+webapp)
launch webservice in intellij as an application
I have access to http://localhost:8080 api+webapp, and if i go to http://localhost:8080/login and press f5 after no issue.

**Comment by OmarHawk** [comment]
> > > @maiido : can you update the test plz? Otherwise the CI will fail
> > 
> > 
> > I have tested this change in a built application and this makes the admin interface unavailable - if we push f5 or use a direct link... so please also make sure that stuff like that will continue to work. ;-)
> 
> Hi, it's strange, I don't have this issue. My test : Compile with maven (profiles dev+webapp) launch webservice in intellij as an application I have access to http://localhost:8080 api+webapp, and if i go to http://localhost:8080/login and press f5 after no issue.

well, but the admin / management interface is behind something like this:

http://localhost:8080/admin/logs

**Comment by maiido** [comment]
@OmarHawk you are right, it's not ok

**Pull request closed without merging**